aten/src/ATen/native/quantized/cpu/qmul.cpp

```diff
@@ -49,24 +49,33 @@ Tensor _mul_out(Tensor& out, const Tensor& self, const Tensor& other) {
 }
 
 template <bool ReLUFused = false>
-Tensor _mul_scalar_out(Tensor& out, const Tensor& self, Scalar other) {
-  int64_t zero_point = out.q_zero_point();
-  double scale = out.q_scale();
-  int64_t self_zero_point = self.q_zero_point();
-  double self_scale = self.q_scale();
-
-  auto iter = TensorIterator::unary_op(out, self);
-  AT_DISPATCH_QINT_TYPES(out.scalar_type(), "qmul", [&]() {
-    cpu_kernel(iter, [&](scalar_t a) -> scalar_t {
-      const auto da = at::dequantize_val(self_scale, self_zero_point, a);
-      float c = da * other.toFloat();
-      if (ReLUFused) {
-        c = std::max<float>(c, 0.0);
-      }
-      return at::quantize_val<scalar_t>(scale, zero_point, c);
+Tensor& _mul_scalar_(Tensor& self, Scalar other) {
+  float other_float = other.toFloat();
+  if (other_float == 0.0f) {
+    // TODO: Replace this when `quantized::zero_` is implemented.
+    auto iter = TensorIterator::unary_op(self, self);
+    AT_DISPATCH_QINT_TYPES(self.scalar_type(), "qmul.scalar.zero", [&]() {
+      cpu_kernel(iter, [&](scalar_t a) -> scalar_t {
+        return static_cast<scalar_t>(self.q_zero_point());
+      });
     });
-  });
-  return out;
+    return self;
+  }
+
+  TORCH_CHECK(other_float > 0,
```

Comment:
Should we support this first prior to landing?

Reply:
I would rather land that, and let someone else take over from this point. Otherwise, this PR will be stuck